rust/ql/test/query-tests/security/CWE-312/test_logging.rs

```diff
@@ -146,6 +147,40 @@ fn test_log(harmless: String, password: String, encrypted_password: String) {
     warn!("message = {}", s2); // (this implementation does not output the password field)
     warn!("message = {:?}", s2); // $ MISSING: Alert[rust/cleartext-logging]=s2
     warn!("message = {:#?}", s2); // $ MISSING: Alert[rust/cleartext-logging]=s2
+
+    let password2 = "123456".to_string(); // Create new password for this test
+
+    // test `log_expect` with sensitive `Option.Some` (which is not output by `log_expect`)
+    let sensitive_opt: Option<String> = Some(password2.clone());
+    let _ = sensitive_opt.log_expect("Option is None");
+
+    // test `log_expect` with sensitive `Result.Ok` (which is not output by `log_expect`)
+    let sensitive_result: Result<String, &str> = Ok(password2.clone());
+    let _ = sensitive_result.log_expect("Result failed");
+
+    // test `log_unwrap` with sensitive `Option.Some` (which is not output by `log_unwrap`)
+    let sensitive_opt2: Option<String> = Some(password2.clone());
+    let _ = sensitive_opt2.log_unwrap();
+
+    // test `log_unwrap` with sensitive `Result.Ok` (which is not output by `log_unwrap`)
+    let sensitive_result2: Result<String, &str> = Ok(password2.clone());
+    let _ = sensitive_result2.log_unwrap();
+
+    // test `log_expect` on `Option` with sensitive message
+    let none_opt: Option<String> = None;
+    let _ = none_opt.log_expect(&format!("Failed with password: {}", password2)); // $ Alert[rust/cleartext-logging]
+
+    // test `log_expect` on `Result` with sensitive message
+    let err_result: Result<String, &str> = Err("");
+    let _ = err_result.log_expect(&format!("Failed with password: {}", password2)); // $ Alert[rust/cleartext-logging]
+
+    // test `log_expect` with sensitive `Result.Err`
+    let err_result2: Result<String, String> = Err(password2.clone());
+    let _ = err_result2.log_expect(""); // $ MISSING: Alert[rust/cleartext-logging]
```

Comment:
The reason we don't find this result is because the model for `clone` assumes `self` would be a reference, but here we do not realize that `password2` is (I believe) implicitly converted to a reference, so we lose the flow.  We've seen similar issues a few times before.